frontend/src/components/mine/Compliance/MineComplianceInfo.js

```diff
@@ -0,0 +1,135 @@
+import moment from "moment";
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { Row, Col } from "antd";
+import { bindActionCreators } from "redux";
+import { connect } from "react-redux";
+
+import Loading from "@/components/common/Loading";
+import NullScreen from "@/components/common/NullScreen";
+import { getMineComplianceInfo } from "@/selectors/complianceSelectors";
+import { fetchMineComplianceInfo } from "@/actionCreators/complianceActionCreator";
+/**
+ * @class  MineTenureInfo - all tenure information related to the mine.
+ */
+
+const propTypes = {
+  mine: PropTypes.object.isRequired,
+  fetchMineComplianceInfo: PropTypes.func.isRequired,
+  mineComplianceInfo: PropTypes.object,
+};
+
+const defaultProps = {
+  mine: {},
+};
+
+export class MineComplianceInfo extends Component {
+  state = { isLoading: true };
+
+  componentDidMount() {
+    this.props
+      .fetchMineComplianceInfo(this.props.mine.mine_detail[0].mine_no)
+      .then(() => this.setState({ isLoading: !this.state.isLoading }));
+  }
+
+  render() {
+    // TODO need to work out how and what is being returned.
```

Comment:
It looks like you've done that? Also, a small note, a `TODO` should have a colon (`TODO: message`). Most syntax highlighters pick it up as a keyword that way

Reply:
I noticed and removed it.